optimus/io/jdbc.py

```diff
@@ -228,31 +246,36 @@ def execute(self, query, limit=None):
         """
 
         # play defensive with a select clause
-        if self.db_driver == "oracle":
-            alias = " t"
-        elif self.db_driver == "presto":
-            alias = ""
+        if self.db_driver == DriverResolver.ORACLE.__str__():
+            query = "(" + query + ") t"
+        elif self.db_driver == DriverResolver.PRESTO.__str__():
+            query = "(" + query + ")"
+        elif self.db_driver == DriverResolver.CASSANDRA.__str__():
+            query = query
         else:
-            alias = " AS t"
-
-        query = "(" + query + self._limit(limit) + ")" + alias
+            query = "(" + query + ") AS t"
 
         logger.print(query)
         logger.print(self.url)
 
         conf = Spark.instance.spark.read \
-            .format("jdbc") \
+            .format("jdbc" if not self.db_driver == DriverResolver.CASSANDRA.__str__() else "org.apache.spark.sql.cassandra") \
             .option("url", self.url) \
-            .option("dbtable", query) \
-            .option("user", self.user)
+            .option("user", self.user) \
+            .option("dbtable", query)
 
-        if self.db_driver != "presto" and self.password is not None:
+        if self.db_driver != DriverResolver.PRESTO.__str__() and self.password is not None:
             conf.option("password", self.password)
 
-        if self.db_driver == "oracle" or self.db_driver == 'postgresql' or self.db_driver == 'presto':
+        if self.db_driver == DriverResolver.ORACLE.__str__() \
+                or self.db_driver == DriverResolver.POSTGRES_SQL.__str__() \
+                or self.db_driver == DriverResolver.PRESTO.__str__():
             conf.option("driver", self.driver_option)
 
-        return conf.load()
+        if self.db_driver == DriverResolver.CASSANDRA.__str__():
```

Comment:
@eschizoid `execute` method returning a dataframe is causing regression, its failing for `df = db_mysql.table_to_df('main', limit='all')` for a connected mysql db object `db_mysql`. See here https://github.com/ironmussa/Optimus/blob/master/tests/sql/sql.ipynb

Reply:
@arpit1997 Unfortunately I can't run MySQL tests since the `requirements.txt` or `requirements-test.txt` does not have the dependency. When I runt this block from the notebook:

```

host = "127.0.0.1"
port= 3306
driver = "mysql"
user = "SA"
password = "SA123"
db_name = "optimus"

db = records.Database('{DRIVER}://{USER}:{PASSWORD}@{HOST}:{PORT}/{DB}'.format(DRIVER=driver, USER=user, PASSWORD=password, HOST=host, PORT=port,DB=db_name))

db.query_file('main.sql')
db.query_file('secondary.sql')
```

I get this error:
---------------------------------------------------------------------------

ModuleNotFoundError                       Traceback (most recent call last)

<ipython-input-27-56e9fc4d31b0> in <module>
      7 db_name = "optimus"
      8 
----> 9 db = records.Database('{DRIVER}://{USER}:{PASSWORD}@{HOST}:{PORT}/{DB}'.format(DRIVER=driver, USER=user, PASSWORD=password, HOST=host, PORT=port,DB=db_name))
     10 
     11 db.query_file('main.sql')

/usr/local/lib/python3.7/site-packages/records.py in __init__(self, db_url, **kwargs)
    260 
    261         # Create an engine.
--> 262         self._engine = create_engine(self.db_url, **kwargs)
    263         self.open = True
    264 

/usr/local/lib/python3.7/site-packages/sqlalchemy/engine/__init__.py in create_engine(*args, **kwargs)
    454     strategy = kwargs.pop("strategy", default_strategy)
    455     strategy = strategies.strategies[strategy]
--> 456     return strategy.create(*args, **kwargs)
    457 
    458 

/usr/local/lib/python3.7/site-packages/sqlalchemy/engine/strategies.py in create(self, name_or_url, **kwargs)
     85                 if k in kwargs:
     86                     dbapi_args[k] = pop_kwarg(k)
---> 87             dbapi = dialect_cls.dbapi(**dbapi_args)
     88 
     89         dialect_args["dbapi"] = dbapi

/usr/local/lib/python3.7/site-packages/sqlalchemy/dialects/mysql/mysqldb.py in dbapi(cls)
    116     @classmethod
    117     def dbapi(cls):
--> 118         return __import__("MySQLdb")
    119 
    120     def on_connect(self):

ModuleNotFoundError: No module named 'MySQLdb'
```

I've tried to install several packages pip packages and none of them seem to work (I am using virtual env with pythbon 3.7)...

If you have any suggestions on how to fix this please legt me know

Reply:
MySqldb might be a lot of pain to install because of system specific dependencies. Please try running for others DBs like postgres, oracle and redshift. I will check things for MySql. 
 Let me know if you run into problems

Reply:
@arpit1997 @Jarrioja setup some database to test .Maybe It could help:

MySQL
Server: 165.227.196.70:3306
Username: test
Password: test
Database: optimus

PostgresSQL
Server: 165.227.196.70:5432
Username: testuser
Password: test
Database: optimus

Redshift
Server: 165.227.196.70:5439
Username: testuser
Password: test
Database: optimus

MS SQL
Server:165.227.196.70:1433
Username: test
Password: test*0261
Database: optimus

Reply:
@arpit1997 Sorry for my late reply works has kept me busy.

Anyways, I tested Redshift and Postgres after my last commit, and the method:

```
df = db_postgres.table_to_df('main', limit='all')
```

is no longer failing, please let me know if there is something else that needs to be done.